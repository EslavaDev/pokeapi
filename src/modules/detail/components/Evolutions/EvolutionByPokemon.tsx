import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {useQuery} from '@tanstack/react-query';
import {evolveRecursion} from '../../utils';
import {pokemonFullDataService} from '@src/services';

export const EvolutionByPokemon = ({url}: any) => {
  const {data: evolution} = useQuery({
    queryKey: ['evolution', url],
    queryFn: pokemonFullDataService(url),
    select: data =>
      data.chain.evolves_to.map((result: any) => [
        data.chain.species.name,
        ...evolveRecursion(result),
      ]),
  });

  return (
    <View style={styles.container}>
      {evolution?.map((item: any, index: number) => (
        <Text key={`${item}-${index}`} style={styles.text}>
          {item?.join(' => ')}
        </Text>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'flex-start',
    paddingHorizontal: 10,
  },
  text: {
    fontSize: 20,
    textTransform: 'capitalize',
  },
});
