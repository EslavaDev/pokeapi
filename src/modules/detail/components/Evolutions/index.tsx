import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {EvolutionByPokemon} from './EvolutionByPokemon';

type Props = {
  evolution: any;
};

export const Evolutions = ({evolution}: Props) => (
  <View style={styles.container}>
    <Text style={styles.text}>Evoluciones</Text>
    <EvolutionByPokemon url={evolution?.evolution_chain?.url} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f0f0f0',
    borderRadius: 10,
    borderColor: '#7a7a7a',
    borderWidth: 1,
    margin: 5,
    padding: 10,
    elevation: 3,
    gap: 15,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    width: '50%',
  },
  text: {
    fontWeight: 'bold',
    fontSize: 25,
    textAlign: 'center',
    textTransform: 'capitalize',
  },
});
