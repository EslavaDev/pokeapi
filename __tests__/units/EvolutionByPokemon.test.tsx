/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: import explicitly to use the types shipped with jest.
import {it, expect} from '@jest/globals';

import renderer from 'react-test-renderer';
import {EvolutionByPokemon} from '@src/modules/detail/components/Evolutions/EvolutionByPokemon';
import {useQuery} from '@tanstack/react-query';

jest.mock('@tanstack/react-query', () => ({
  useQuery: jest.fn(),
}));

it('should render the component without crashing', () => {
  const mockData = {
    data: [
      ['Bulbasaur', 'Ivysaur', 'Venusaur'],
      ['Charmander', 'Charmeleon', 'Charizard'],
      ['Squirtle', 'Wartortle', 'Blastoise'],
    ],
  };

  useQuery.mockImplementation(() => mockData);
  const component = renderer.create(
    <EvolutionByPokemon url="https://pokeapi.co/api/v2/pokemon/1" />,
  );

  // Assert that the component renders without crashing
  expect(component.toJSON()).toMatchSnapshot();
});
