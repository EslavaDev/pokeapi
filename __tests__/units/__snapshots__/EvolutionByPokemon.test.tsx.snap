// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component without crashing 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "textTransform": "capitalize",
      }
    }
  >
    Bulbasaur =&gt; Ivysaur =&gt; Venusaur
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "textTransform": "capitalize",
      }
    }
  >
    Charmander =&gt; Charmeleon =&gt; Charizard
  </Text>
  <Text
    style={
      {
        "fontSize": 20,
        "textTransform": "capitalize",
      }
    }
  >
    Squirtle =&gt; Wartortle =&gt; Blastoise
  </Text>
</View>
`;
